server/src/rest/api.rs

```diff
@@ -0,0 +1,1370 @@
+    Extension(mut _grpc_clients): Extension<GrpcClients>,
+    Path(operator_id): Path<String>,
+) -> Result<Json<Operator>, (StatusCode, String)> {
+    rest_debug!("get_operator({})", operator_id);
+    if !is_uuid(&operator_id) {
+        return Err((StatusCode::BAD_REQUEST, "Invalid operator id".to_string()));
+    }
+    // Get Client
+    // TODO let _client_option = grpc_clients.storage.get_client().await;
+    // if client_option.is_none() {
+    //     let error_msg = "svc-storage unavailable.".to_string();
+    //     rest_error!("(get_operator) {}", &error_msg);
+    //     return Err((StatusCode::SERVICE_UNAVAILABLE, error_msg));
+    // }
+    // let mut client = client_option.unwrap();
+    //TODO
+    Ok(Json(Operator::random()))
+}
+
+//-----------------------------------------------------------
+// R2 DEMO
+//-----------------------------------------------------------
+
+#[utoipa::path(
+    get,
+    path = "/assets/demo/aircraft",
+    tag = "svc-assets",
+    responses(
+        (status = 200, description = "Assets successfully found", body = [Aircraft]),
+        (status = 503, description = "Could not connect to other microservice dependencies")
+    ),
+)]
+/// Get all aircraft from the database.
+pub async fn get_all_aircraft(
+    Extension(mut grpc_clients): Extension<GrpcClients>,
+) -> Result<Json<Vec<Aircraft>>, (StatusCode, String)> {
+    let filter = AdvancedSearchFilter::search_is_not_null(String::from("deleted_at"));
+    let vehicle_client_option = grpc_clients.storage_vehicle.get_client().await;
+    if vehicle_client_option.is_none() {
+        let error_msg = "svc-storage unavailable.".to_string();
+        rest_error!("(get_all_aircraft) {}", &error_msg);
+        return Err((StatusCode::SERVICE_UNAVAILABLE, error_msg));
+    }
+
+    let mut vehicle_client = vehicle_client_option.unwrap();
+
+    let vehicle_response = vehicle_client.search(filter.clone()).await;
+
+    if vehicle_response.is_err() {
+        let error_msg = "svc-storage unavailable.".to_string();
+        rest_error!("(get_all_aircraft) {}", &error_msg);
+        return Err((StatusCode::SERVICE_UNAVAILABLE, error_msg));
+    }
+    let mut vehicles = vehicle_response.unwrap().into_inner().list;
+
+    let mut assets = Vec::new();
+
+    for vehicle in vehicles.drain(..) {
+        assets.push(Aircraft::from(vehicle).unwrap());
+    }
+
+    Ok(Json(assets))
+}
+
+#[utoipa::path(
+    get,
+    path = "/assets/demo/vertiports",
+    tag = "svc-assets",
+    responses(
+        (status = 200, description = "Assets successfully found", body = [Vertiport]),
+        (status = 503, description = "Could not connect to other microservice dependencies")
+    ),
+)]
+/// Get all vertiports from the database.
+pub async fn get_all_vertiports(
+    Extension(mut grpc_clients): Extension<GrpcClients>,
+) -> Result<Json<Vec<Vertiport>>, (StatusCode, String)> {
+    let filter = AdvancedSearchFilter::search_is_not_null(String::from("deleted_at"));
```

Comment:
entry info!

Reply:
check for the other api functions as well

Reply:
[4428782](https://github.com/Arrow-air/svc-assets/pull/16/commits/4428782e670f74e05fb753342bf5365399d11255)